### DISCUSSION TITLE
Understanding INPUT_OBJECT level custom directives

### DISCUSSION THREAD

**Discussion by justinsumida** [graphql-java/graphql-java#3418] [Q&A]
I am trying to implement an INPUT_OBJECT type directive to do some validation that 2 of 3 inputs are supplied on certain inputs. The problem I am running into is that using SchemaDirectiveWiring.onInputObjectType() only fires on the application starts. I also am not sure how to gain visibility into what fields were actually supplied in GraphQLInputObjectType to validate that 2 of 3 inputs were supplied.
Has anyone attempted to make an Input Object level directive before? 

I am using version 17.3

**Comment by bbakerman** [comment]
Yeah your challenge is that you want to act on the directive at runtime - eg during runtime.

The main callback at runtime we have is of course the `DataFetcher`  - it can look at input args and their types and act on them

In fact the library https://github.com/graphql-java/graphql-java-extended-validation uses this extensively to validate that input objects are the right shape.

if you look at this code you will see lots of examples making input object directives - who knows, it might even do what you are planning to do
